src/docs/truffle/reference/contract-abstractions.md

```diff
@@ -0,0 +1,278 @@
+---
+title: "Truffle | Contract Abstractions"
+layout: "docs.hbs"
+---
+# Contract Abstractions
+
+Truffle provides contract abstractions for interacting with your contracts.  Skip ahead to the [api section](/docs/truffle/reference/contract-abstractions#api) for a list of contract methods.
+
+### Usage
+
+To obtain a contract abstraction you can require it with the contract name from the `artifacts` object.  Outside of the console this is an object available in migration files, tests, and exec scripts.
+You would require your contract as follows:
+
+```javascript
+const MyContract = artifacts.require("MyContract");
+```
+
+You can also obtain one in the developer console.  Your contract types are available here and all you need to do is use the `at`, `deployed`, or `new` method.
+
+```javascript
+truffle(development)> const myContract = await MyContract.deployed();
+```
+
+You now have access to the following functions on `MyContract`, as well as many others:
+
+* `at()`: Create an instance of `MyContract` that represents your contract at a specific address.
+* `deployed()`: Create an instance of `MyContract` that represents the default address managed by `MyContract`.
+* `new()`: Deploy a new version of this contract to the network, getting an instance of `MyContract` that represents the newly deployed instance.
+
+Each instance is tied to a specific address on the Ethereum network, and each instance has a 1-to-1 mapping from Javascript functions to contract functions. For instance, if your Solidity contract had a function defined `someFunction(uint value) {}` (solidity), then you could execute that function on the network like so:
```

Comment:
I'm a bit wary of potentially confusing language around the term `instance`, since it might mean "abstraction instance" vs. "contract instance". 

"Create an instance" in particular seems like it might be misleading.

But maybe not a big deal. `MyContract.at()` in name is clear on its own.